Fix Bug exposed by Fuzz Tests

**Describe the bug**
Fuzz testing is failing with libcst errors I don't get.

```
fuzz run-test: commands[2] | coverage run fuzz.py
<string>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
<string>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
<string>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
<string>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
<string>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
<string>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
Highest target scores:
               5  (label='(hypothesmith) number of unique ast node types')
               9  (label='(hypothesmith) instructions in bytecode')
              10  (label='(hypothesmith from_node) number of unique ast node types')
              13  (label='(hypothesmith) total number of ast nodes')
              35  (label='(hypothesmith from_node) total number of ast nodes')
              48  (label='(hypothesmith from_node) instructions in bytecode')

Traceback (most recent call last):
  File "<string>", line 7, in __init__
  File "/home/cooper/repos/black/.tox/fuzz/lib/python3.9/site-packages/libcst/_nodes/base.py", line 115, in __post_init__
    self._validate()
  File "/home/cooper/repos/black/.tox/fuzz/lib/python3.9/site-packages/libcst/_nodes/expression.py", line 1148, in _validate
    raise CSTValidationError(
  File "<string>", line None
libcst._nodes.base.CSTValidationError: Must have at least one space around comparison operator.
```

**To Reproduce** 
Run `tox -e fuzz` on head

**Expected behavior**
Fuzz tests to pass.

These have now become flaky, the error looks like a real problem tho:
(Have seen it with Python 3.6 and 3.9)
```
AssertionError: INTERNAL ERROR: Black produced different code on the second pass of the formatter.  Please report a bug on https://github.com/psf/black/issues.  This diff might be helpful: /tmp/blk__o2fa65u.log
```
Example: https://github.com/psf/black/pull/1918/checks?check_run_id=1679427463

--- a/src/black/__init__.py
+++ b/src/black/__init__.py
@@ -2635,7 +2635,7 @@ def list_comments(prefix: str, *, is_endmarker: bool) -> List[ProtoComment]:
     consumed = 0
     nlines = 0
     ignored_lines = 0
-    for index, line in enumerate(prefix.split("\n")):
+    for index, line in enumerate(re.split("\r?\n", prefix)):
         consumed += len(line) + 1  # adding the length of the split '\n'
         line = line.lstrip()
         if not line:

--- a/tests/test_black.py
+++ b/tests/test_black.py
@@ -1793,6 +1793,12 @@ class BlackTestCase(BlackBaseTestCase):
         finally:
             os.chdir(str(old_cwd))
 
+    def test_newline_comment_interaction(self) -> None:
+        # Source: https://github.com/psf/black/pull/1958/checks?check_run_id=1849391379#step:5:72
+        source = "class A:\\\r\n# type: ignore\n pass\n"
+        output = black.format_str(source, mode=DEFAULT_MODE)
+        black.assert_stable(source, output, mode=DEFAULT_MODE)
+
 
 with open(black.__file__, "r", encoding="utf-8") as _bf:
     black_source_lines = _bf.readlines()

--- a/tests/test_black.py
+++ b/tests/test_black.py
@@ -1794,7 +1794,6 @@ class BlackTestCase(BlackBaseTestCase):
             os.chdir(str(old_cwd))
 
     def test_newline_comment_interaction(self) -> None:
-        # Source: https://github.com/psf/black/pull/1958/checks?check_run_id=1849391379#step:5:72
         source = "class A:\\\r\n# type: ignore\n pass\n"
         output = black.format_str(source, mode=DEFAULT_MODE)
         black.assert_stable(source, output, mode=DEFAULT_MODE)

